libpod/container_internal.go

```diff
@@ -976,6 +976,19 @@ func (c *Container) completeNetworkSetup() error {
 			}
 		}
 	}
+	// check if we have a bindmount for /etc/hosts
+	hostsBindMount := state.BindMounts["/etc/hosts"]
+	if hostsBindMount != "" && len(c.cniHosts()) > 0 {
+		// read the existing hosts
+		b, err := ioutil.ReadFile(hostsBindMount)
+		if err != nil {
+			return err
+		}
```

Comment:
if /etc/hosts doesn't exist, do we want to fail or just continue on?

Reply:
If this fails, this means that the system is going to fail later when it attempts to mount the file.  So yes we want to fail

Reply:
Should error - something has gone seriously wrong in container creation if BindMounts refers to a path that does not exist.